test/tests/domains-task.js

```diff
@@ -0,0 +1,118 @@
+"use strict";
+
+const requireInject = require("require-inject");
+const {expect} = require("chai");
+
+const {
+  REQUEST_DOMAINS_UPDATE
+} = require("../../src/lib/background/state/types/domains");
+
+const TEST_PATH = "../../src/lib/background/domains/task";
+
+function makeTestModule({dispatch, lastUpdated})
+{
+  lastUpdated = lastUpdated || 0;
+
+  let deps = {
+    "global/window":
+    {
+      setInterval() {},
+      Date
+    },
+    "../../src/lib/background/state":
+    {
+      store:
+      {
+        dispatch: dispatch || (() => {})
+      }
+    },
+    "../../src/lib/common/env/chrome":
+    {
+      chrome:
+      {
+        runtime: {},
+        storage:
+        {
+          local:
+          {
+            get(name, callback)
+            {
+              if (name == "domains.lastUpdated")
+                return callback({"domains.lastUpdated": lastUpdated});
+              throw new Error("trying to get an unknown setting.");
+            },
+            set(settings, callback)
+            {
+              if (settings["domains.lastUpdated"])
+              {
+                lastUpdated = settings["domains.lastUpdated"];
+                callback();
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  let settings = requireInject("../../src/lib/common/settings", deps);
+
+  let testModule = requireInject(TEST_PATH, deps);
+
+  return Object.assign({}, testModule, {settings});
```

Comment:
Detail: This is pretty much the same as writing `testModule.settings = settings` so if we were to export "settings" from that module we'd be unknowingly overwriting it here.

It's not an issue at this point but it may become one if we use this approach more widely as seen in test/tests/saga-updateDomainsList.js.